source/plugins/go/src/telemetry.go

```diff
@@ -500,6 +537,14 @@ func PushToAppInsightsTraces(records []map[interface{}]interface{}, severityLeve
 			populateKubeMonAgentEventHash(record, ConfigError)
 		} else if strings.Contains(logEntry, "E! [inputs.prometheus]") {
 			populateKubeMonAgentEventHash(record, PromScrapingError)
+		} else if strings.Contains(logEntry, "Lifetime validation failed. The token is expired.") {
```

Comment:
Are we covering all the possible error scenarios of the MDSD here? Can you please look at CI agent log dashboards and telemetry to make sure we have metric for all key error scenarios?

Reply:
Added a couple more but these were the only error types I found...Once we convert these to metrics will again review and keep on adding more scenarios if found. Let me know if you think otherwise